cli.sh

```diff
@@ -978,91 +972,90 @@ add_shared_hook_repo() {
 #   the global or local list.
 #####################################################
 remove_shared_hook_repo() {
-    SET_SHARED_GLOBAL=
+    SET_SHARED_TYPE="--shared"
     SHARED_REPO_URL=
 
-    case "$1" in
-    "--global")
-        SET_SHARED_GLOBAL=1
+    if echo "$1" | grep -qE "\-\-(shared|local|global)"; then
+        SET_SHARED_TYPE="$1"
         SHARED_REPO_URL="$2"
-        ;;
-    "--local")
-        SET_SHARED_GLOBAL=
-        SHARED_REPO_URL="$2"
-        ;;
-    *)
+    else
         SHARED_REPO_URL="$1"
-        ;;
-    esac
+    fi
 
     if [ -z "$SHARED_REPO_URL" ]; then
-        echo "! Usage: \`git hooks shared remove [--global|--local] <git-url>\`" >&2
+        echo "! Usage: \`git hooks shared remove [--shared|--local|--global] <git-url>\`" >&2
         exit 1
     fi
 
-    if [ -n "$SET_SHARED_GLOBAL" ]; then
-        CURRENT_LIST=$(git config --global --get githooks.shared)
-        NEW_LIST=""
+    if [ "$SET_SHARED_TYPE" != "--shared" ]; then
 
-        IFS="$IFS_COMMA_NEWLINE"
+        if [ "$SET_SHARED_TYPE" = "--local" ] && ! is_running_in_git_repo_root; then
+            echo "! The current directory \`$(pwd)\` does not" >&2
+            echo "  seem to be the root of a Git repository!" >&2
+            exit 1
+        fi
+
+        CURRENT_LIST=$(git config "$SET_SHARED_TYPE" --get-all githooks.shared)
+
+        # Unset all and add them back
+        git config "$SET_SHARED_TYPE" --unset-all githooks.shared
 
+        IFS="$IFS_NEWLINE"
```

Comment:
Can't `$CURRENT_LIST` be a comma-separated list at this point? (from a previous version)

Reply:
Not it cant, because when you updated (and the legacy transform `legacy_transform_split_global_shared_entries` worked correctly) , you will have multiple values in the git config and with IFS_NEWLINE we split them apart.